Sources/UserDefaultsGeneratorCore/Generator.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  Generator.swift
+//  CYaml
+//
+//  Created by Yudai Hirose on 2019/09/18.
+//
+
+import Foundation
+
+public protocol Generator {
+    func generate() throws
+}
+
+public struct GeneratorImpl: Generator {
+    let configurations: [Configuration]
+    let outputPath: URL
+    public init(configurations: [Configuration], outputPath: URL) {
+        self.configurations = configurations
+        self.outputPath = outputPath
+    }
+    public func generate() throws {
+        let content = enumDefinition() + "\n" + userDefaultsExtensions()
+        try content.write(to: outputPath, atomically: true, encoding: .utf8)
+    }
+    
+    func enumDefinition() -> String {
```

Comment:
```suggestion
    private func enumDefinition() -> String {
```

Reply:
In next PR, to write unit test for enumDefinition. So, I decide to keep internal access control.